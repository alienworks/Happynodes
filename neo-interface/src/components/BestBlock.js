
import React, { Component } from 'react'
import { connect } from 'react-refetch'
import config from './config'

class bestblock extends Component {
    render() {
        const {bestblock} = this.props
        if (bestblock.pending) {
            return (
               <p></p>
            );
        }
        if (bestblock.value === null || bestblock.value === 'undefined'){
            return (
                <div className="top-block blue">
                <h2>42</h2>
                <p>BEST BLOCK</p></div>
            );
        }
        return (
            <div className="top-block blue">
            <h2>{Number(bestblock.value.bestblock).toLocaleString()}</h2>
            <p>BEST BLOCK</p></div>
        );
    }
}
  
// export default bestblock;

export default connect( (props)=> ({
    bestblock: {
        url: config.api_url.concat(`/bestblock`),
        refreshInterval: 3000
    }
}))(bestblock)
