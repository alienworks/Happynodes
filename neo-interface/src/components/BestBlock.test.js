import React from 'react';
import ReactDOM from 'react-dom';
import BestBlock from './BestBlock';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BestBlock />, div);
    ReactDOM.unmountComponentAtNode(div);
});


describe("test against valid snapshot", ()=> {
    it("redner as expected", ()=> {
        const tree = renderer.create(<BestBlock bestblock={{value:{bestblock:1234}}} />).toJSON();

        expect(tree).toMatchSnapshot();

    });
});
  