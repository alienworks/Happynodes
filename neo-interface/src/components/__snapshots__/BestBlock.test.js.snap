// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test against valid snapshot redner as expected 1`] = `
<div
  className="top-block blue"
>
  <h2>
    1,234
  </h2>
  <p>
    BEST BLOCK
  </p>
</div>
`;
